external_parser/parse_example_binary.cc

```diff
@@ -209,12 +172,15 @@ bool binary_parser::read_checkpoint_msg(io_buf *input) {
   _example_joiner->set_problem_type_config(
       checkpoint_info->problem_type_config());
 
+  _got_checkpoint_info = true;
   return true;
 }
 
 bool binary_parser::read_regular_msg(io_buf *input,
```

Comment:
instead of passing a bool here could we potentially add a method `bool checkpoint_info_ready()` and call it after calling `read_regular_msg()` ?

Reply:
Good point. `i_joiner` should hold that info since it might have being externally configured prior to the construction of the parser object.

I rather keep the checking side of the code as is today so we entirely avoid parsing the examples in such case.